### PR TITLE
fix issue #167

### PR DESCRIPTION
**Pull request by valq7711**
Clear tables aliases before each request
**requires the last version of pyDAL**  (with [this PR](https://github.com/web2py/pydal/pull/630) included)

### CODE REVIEW

File: py4web/core.py [hunk 1]
```diff
@@ -272,6 +272,7 @@ def on_success(self):
 class DAL(pydal.DAL, Fixture):
     def on_request(self):
         threadsafevariable.ThreadSafeVariable.restore(ICECUBE)
+        self._aliased_tables.clear() # should we do tha same on_error/on_success?
```
Comment (mdipierro) [inline py4web/core.py:275]: Damn if we need to clear it, _aliased_tables is not thread safe. Need to check before merging this.

Comment (valq7711) [reply py4web/core.py:275]: Oh yeah, my bad, is it fixed now? (with [below commit](https://github.com/web2py/py4web/pull/207/commits/d82afd1df460cf0650bda88c46af343e633abfbb))

Comment (mdipierro) [reply py4web/core.py:275]: testing a fix

**Comment by mdipierro** [comment]
I think this is now fixed in pydal. Can you please confirm?

**Comment by valq7711** [comment]
Yes, now it works just fine! Thanks a lot!

**Comment by jpsteil** [comment]
I'm still seeing an issue.  I have the following form:

```
form = Form([Field('search', length=50, default=search_filter),
         Field('active', 'boolean', label='Active Only', default=active_filter),
         Field('test_date', 'date', requires=IS_DATE('%m/%d/%Y'))],
        keep_values=True, formstyle=FormStyleBulma)
```
In the screenshot you can see it is giving an error even though it is displaying it correctly.  From my reading this is because the HTML5 date control displays your date in your current locale format but always deals with it in the form field as YYYY-MM-DD format.  The pydal validator wants to read it in the format that you specify on the IS_DATE validator.  I hope I'm making sense.

-Jim

Screenshot from 2020-05-26 16-42-48

**Pull request closed without merging**